src/extensions/replay/sessionrecording.ts

```diff
@@ -1198,6 +1225,34 @@ export class SessionRecording {
         }
     }
 
+    private _pauseRecording() {
+        if (this.status === 'paused') {
+            return
+        }
+
+        this._urlBlocked = true
+
+        this.clearBuffer()
+        clearInterval(this._fullSnapshotTimer)
+        this._tryAddCustomEvent('recording paused', { reason: 'url blocker' })
```

Comment:
i think the _urlBlocked above means we don't get this custom event because we return early 

so, we could add the custom event and then set the flag... 

makes me wonder about the buffer clear here as well
i don't know that we have a good way to do this but... we're dropping unemitted data here without knowing it's to do with the blocked page maybe

so i'm thinking what will happen is

* add some data to the buffer by moving their mouse and clicking a button
* it isn't big or old enough so we don't send it
* user navigates to the blocked page because of the button click
* we clear the buffer and lose the mouse data we didn't send yet

i don't think we have a good way of doing this but in a perfect world we'd send all of the data before the blocked page in the buffer and clear everything after it

(should err on the side of caution while we don't have a good way of being sure though but would be nice to lose as little as possible)

Reply:
Asked a question re the custom event in a different thread, but I've changed to around, and from my testing it was reliably sending all events before hitting the block condition.